src/core_plugins/elasticsearch/lib/cluster.js

```diff
@@ -0,0 +1,137 @@
+import elasticsearch from 'elasticsearch';
+import { get, set, isEmpty, cloneDeep, pick } from 'lodash';
+import toPath from 'lodash/internal/toPath';
+import Boom from 'boom';
+
+import filterHeaders from './filter_headers';
+import { parseConfig } from './parse_config';
+
+export default class Cluster {
+  constructor(config) {
+    this._config = Object.assign({}, config);
+    this.errors = elasticsearch.errors;
+
+    createClients.call(this);
+
+    return this;
+  }
+
+  /**
+   * callWithRequest
+   *
+   * Performs a call to ES, passing through whitelisted headers in the request
+   *
+   * @param {Object|undefined} req - The request object
+   * @param {string} endpoint
+   * @param {Object} clientParams
+   * @param {Object} options
+   * @param {boolean} options.wrap401Errors
+   * @returns {Promise}
+   */
+  callWithRequest = (req = {}, endpoint, clientParams = {}, options = {}) => {
```

Comment:
The arrow functions give us the the lexical scoping of _this_, allowing for the destructuring assignment of the functions. Without this, simply calling callWithRequest() without the object would result in _this_ being undefined in strict mode.

We can use normal function declarations, but would need to update the calls to be made on the cluster object instead of destructuring the assignment of the functions themselves.

Preferences? Or is there anything option I am not missing?